pep-0561.rst

```diff
@@ -42,40 +43,73 @@ checkers to fail on packages that are highly dynamic (e.g. sqlalchemy
+syntax.
+
+"stubs" - files containing only type information, empty of runtime code.
+
+"Distributions" are the packaged files which are used to publish and distribute
+a release. [3]_
+
+"Module" a file containing Python runtime code or stubbed type information.
+
+"Package" a directory or directories that namespace Python modules.
+
+
 Specification
 =============
 
 There are several motivations and methods of supporting typing in a package.
-This PEP recognizes three (3) types of packages that may be created:
+This PEP recognizes three (3) types of packages that users of typing wish to
+create:
 
 1. The package maintainer would like to add type information inline.
 
 2. The package maintainer would like to add type information via stubs.
 
-3. A third party would like to share stub files for a package, but the
-   maintainer does not want to include them in the source of the package.
+3. A third party or package maintainer would like to share stub files for
+   a package, but the maintainer does not want to include them in the source
+   of the package.
    
 This PEP aims to support these scenarios and make them simple to add to
 packaging and deployment.
 
 The two major parts of this specification are the packaging specifications
-and the resolution order for resolving module type information. The packaging
-spec is based on and extends PEP 345 metadata. The type checking spec is
-meant to replace the ``shared/typehints/pythonX.Y/`` spec of PEP 484 [2]_.
+and the resolution order for resolving module type information. The type
+checking spec is meant to replace the ``shared/typehints/pythonX.Y/`` spec
+of PEP 484 [2]_.
+
+New third party stub libraries SHOULD distribute stubs via the third party
+packaging methods proposed in this PEP in place of being added to typeshed.
+Typeshed will remain in use, but if maintainers are found, third party stubs
+in typeshed MAY be split into their own package.
 
-New third party stub libraries are encouraged to distribute stubs via the
-third party packaging proposed in this PEP in place of being added to 
-typeshed. Typeshed will remain in use, but if maintainers are found, third
-party stubs in typeshed are encouraged to be split into their own package.
 
 Packaging Type Information
 --------------------------
+
 In order to make packaging and distributing type information as simple and
-easy as possible, the distribution of type information, and typed Python code
-is done through existing packaging frameworks. This PEP adds a new item to the
-``*.distinfo/METADATA`` file to contain metadata about a package's support for
-typing. The new item is optional, but must have a name of ``Typed`` and have a
-value of either ``inline`` or ``stubs``, if present.
+easy as possible, packaging and distribution is done through existing
+frameworks.
+
+To enable the safe distribution of typing information, packaging tools MUST
+install a ``.typeinfo`` file per installed package that indicates it supports
```

Comment:
Some Windows tools have problems with dotfiles, so `py.typeinfo` would likely be a better filename for this purpose.

You should also be explicit that for namespace package portions, the `py.typeinfo` file would go inside the namespace submodule, not at the namespace package level.

Reply:
Ah, if the onus is on the maintainer to create the file, then yes, this would be an annoyance. 👍 the name sounds good.